security/src/main/java/io/micronaut/security/authentication/ServerAuthentication.java

```diff
@@ -34,6 +35,7 @@
  * @author James Kleeh
  * @since 3.0.0
  */
+@ReflectiveAccess
 public class ServerAuthentication implements Authentication {
```

Comment:
`@Reflective` adds the class to `reflect-config.json`, but it only adds the class. 

`@TypeHint(ServerAuthentication::class, accessType = [ALL_PUBLIC])` adds more information because it declares all public methods/constructors.

@morki did you try this patch in your code and did it work?

Reply:
@morki it would be good to have a sample application which reproduces the issue so that we can added to our graalvm test suite.

Reply:
@ilopmar previously I didn't test but now I did and replaced `@ReflectiveAccess` with `@Introspected` (possibly better by reducing reflection and for future serdes module)

@sdelamo we have sample application in graal-tests which needs to be just simply modified to show this behaviour: https://github.com/micronaut-graal-tests/micronaut-security-session-graal

Append to `application.yml`:

```yaml
---
micronaut:
  session:
    http:
      redis:
        enabled: true

redis:
  url: redis://127.0.0.1:6379
```

Add this dependency to `build.gradle`:

```kotlin
    implementation("io.micronaut.redis:micronaut-redis-lettuce")
```

But I don't know how to setup Redis for this test.

Reply:
I can modify the app and our test suite to test security-session always with Redis instead of in-memory